SQL.Rmd

```diff
@@ -0,0 +1,1190 @@
+
+```sql
+SELECT some_column, another_column, ...
+FROM my_table
+  (SELECT some_column, another_column, ...
+   FROM my_table
+   WHERE condition) AS subquery_name
+WHERE condition(s);
+```
+
+#### Inside WHERE
+
+* Filters results, based on data that would otherwise have to be created separately
+* Can create a list for filtering
+* Returns only a single column
+
+```sql
+SELECT some_column, another_column, ...
+FROM my_table
+WHERE condition(s)
+  (SELECT some_column, another_column, ...
+   FROM my_table
+   WHERE condition);
+```
+
+This is most common type of subquery so let's get practical for this one.
+
+Find the records where the length of stay (`LengthOfStay`) is more than the average for `HospitalCode` "B120H". 
+
+```{sql sub-where, connection="conn", exercise=TRUE}
+
+```
+
+```{sql sub-where-solution}
+SELECT *
+FROM borders_data
+WHERE LengthOfStay > 
+    (SELECT AVG(LengthOfStay)
+    FROM borders_data
+    WHERE HospitalCode = "B120H")
+    AND HospitalCode = "B120H";
+```
+
+
+## Table Management
+
+This course has taken a pretty wide view at interacting with existing tables in databases. Using these as building blocks, the power to build all sorts of queries in SQL is in your hands. However, what about changing the tables in some way, either creating new ones, editing, or deleting? Well, this all falls under an acronym (and we love an acronym) to describe operations, [**CRUD**](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) (create, read, update, and delete). 
+
+Databases that we use most likely have access restrictions to prevent issues with our live data. However, other use cases exist where we create a view for ourselves or we may even have the need to utilise a database for a bespoke project. This section explores the creation, updating, and deleting of tables within a database. Just to reiterate, there's a lot of supporting theory around database design and optimal database solutions, this course doesn't cover this in any detail but resources will be made available if this is something you're interested in.
+
+*Most of the exercises in this section don't have output, you can still build a query to view the table after to check the results though. However, you can only run one query at a time.*
+
+### CREATE TABLE
+
+We know that a table in a database has a particular structure with constraints in place, this is an explicit step in the creation of the table. The structure is defined by its [table schema](https://www.sqlite.org/schematab.html). As part of this, each table has:
+
+* A name (ideally in the singular with no spaces)
+* Date type
+* Constraints (*optional*)
+* Default values (*optional*)
+
+*If a table already exists with the same name, SQL will throw an error. You can suppress the error and stop the creation of the table using the `IF NOT EXISTS` clause.*
+
+```sql
+CREATE TABLE (IF NOT EXISTS) new_table (
```

Comment:
The brackets round IF NOT EXISTS threw me in the later exercise, should we take those out?